ReactNativeClient/lib/services/SearchEngine.js

```diff
@@ -235,19 +236,39 @@ class SearchEngine {
 		return occurenceCount / spread;
 	}
 
-	processResults_(rows, parsedQuery) {
+	processBasicSearchResults_(rows, parsedQuery) {
+		const valueRegexs = parsedQuery.keys.includes('_') ? map(parsedQuery.terms['_'], 'valueRegex') : [];
+
 		for (let i = 0; i < rows.length; i++) {
 			const row = rows[i];
-			const offsets = row.offsets.split(' ').map(o => Number(o));
-			row.weight = this.calculateWeight_(offsets, parsedQuery.termCount);
-			row.fields = this.fieldNamesFromOffsets_(offsets);
+			const testTitle = regex => new RegExp(regex, 'ig').test(row.title);
+			const matchedFields = {
+				title: parsedQuery.keys.includes('title') || valueRegexs.some(testTitle),
+				body: parsedQuery.keys.includes('body') || valueRegexs.some(negate(testTitle)),
```

Comment:
Why not have 'body' in the fields irrespective of whether the regex contains the title.
`body: true`

This would make my test case pass.

If the word was not in the body, it is not displayed. No harm done.
 Else, it works! Is there a performance penalty?

Reply:
Nice idea. Haven't performed a performance test but maybe we can only assume  keywords always appearing in body when `rows.length <= 15` to avoid performance penalty?

Reply:
there is already a limit of 20 in GotoAnything.jsx

Reply:
Really? see this image, it seems that there are 141 search results to process
<img width="466" alt="图片" src="https://user-images.githubusercontent.com/12264626/81362111-a6acc200-9112-11ea-9cff-1e5873e072b1.png">

Reply:
I meant that the expensive body regex search has a limit of 20 in GotoAnything.

Reply:
😯 that's right. And I test with 1000 notes, it seems that everything's fine. So I push a commit now